### PR TITLE
Singly Linked List - Python

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented all basic features of Singly Linked List in Python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: Python/ds/SinglyLinkedList.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    def deleteNodeAtPosition(self, index):
+        if self.head == None:
+            print("Empty list, nothing to delete!")
+            return
+        pos = self.head
+        if index == 1:
+            self.head = pos.next
+            temp = None
+            return
+        for i in range(1,index-1):
```
Comment (SKAUL05) [inline Python/ds/SinglyLinkedList.py]: Use meaningful and readable variable names instead of `i`.

File: Python/ds/SinglyLinkedList.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    print("7. Print the List")
+    print("8. Count nodes in the List")
+    print("9. Exit")
+    print("--Use 1-based indexing--")
+    while(True):
+        print("Enter your choice: ")
+        ch = int(input())
+        if ch == 1:
+            print("Enter the element to insert: ")
+            ndata = int(input())
```
Comment (SKAUL05) [inline Python/ds/SinglyLinkedList.py]: Pass output statements in `input()`
Like:
`ndata = int(input("Enter the element to insert: "))`

Comment (SangeetaMishra) [reply Python/ds/SinglyLinkedList.py]: I have changed it to 'ind'

**Review by SKAUL05 — APPROVED** [review]

**Review by Kajol-Kumari — COMMENTED** [review]
Programs works fine and LGTM :)

Please add one example test case input and its output in the comment form below the code section, it would be better for the reader to understand. For this particular code, try to run all the conditions and paste the output below the code in commented form.

**Comment by SangeetaMishra** [comment]
> Programs works fine and LGTM :)
> 
> Please add one example test case input and its output in the comment form below the code section, it would be better for the reader to understand. For this particular code, try to run all the conditions and paste the output below the code in commented form.

I have added the sample input and output.

**Review by SKAUL05 — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
Perfect! 
This can be merged now @HarshCasper

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
I would like you to add a simple yet effective feature,

would you write some script that can take an array input from the user and convert that to a linked list instead of inserting some nodes step by step?

That will save time right.

**Review by vatsalkesarwani12 — APPROVED** [review]
Code working fine

**Comment by SangeetaMishra** [comment]
> I would like you to add a simple yet effective feature,
> 
> would you write some script that can take an array input from the user and convert that to a linked list instead of inserting some nodes step by step?
> 
> That will save time right.

I personally don't feel that using an array would be a better option. It might save time but the purpose of this code is to use Linked List.

**Pull request merged**